packages/corelib/src/playout/__tests__/infinites.test.ts

```diff
@@ -23,6 +23,10 @@ describe('Infinites', () => {
 		clearOrAdlib?: boolean | number,
 		infinite?: PieceInstance['infinite']
 	): PieceInstance {
+		let dynamicallyInserted = undefined
+		if (clearOrAdlib) {
+			dynamicallyInserted = clearOrAdlib === true ? { time: Date.now() } : { time: clearOrAdlib }
```

Comment:
This is very confusing. `clearOrAdlib` is treated as a boolean both in the if-statement but also in the ternary conditional. But in the else-clause of the ternary it is all of a sudden a `Time`??

Reply:
Also, in theory both the conditional in the if-statment and the tenary is the same, so you should never hit the else in the ternary. But since the type is `boolean | number` who knows what can happen.

Sidenote: In what scenario does it make logical sense to have a value to be able to be both a boolean and a number?? (I know you didn't write this, so not aiming that question at you)

Reply:
> clearOrAdlib is treated as a boolean both in the if-statement but also in the ternary

first it's checked whether it's `true | number`, then in the ternary it's narrowed down to `true` or `number`. You can see how narrowing down progresses when you hover over the variable in different places.
![image](https://user-images.githubusercontent.com/9103996/204546156-3a1dfad3-eaf2-4fc1-9298-dbd5c00fdd62.png)
![image](https://user-images.githubusercontent.com/9103996/204546979-74966148-7b9c-4743-9ed6-3e758a0a04a1.png)

> In what scenario does it make logical sense to have a value to be able to be both a boolean and a number

It's like a function overload, but without the extra signatures. Not ideal, but not completely illogical.

Reply:
>  but not completely illogical.

But also not logical? ;) 

Thanks for the answer. Personally I think we should try and stay away from "mixing" simple types like this. For one, they become extremely difficult to find a good name for, and we know what that indicates, (if you knew nothing about this code and saw a variable called `clearOrAdlib`, what would you expect the type to be? - My guess would be neither a boolean nor a number...) and it seems to rely on the fact that JavaScript doesn't have types at runtime